src/components/table/column-picker/column-picker.component.ts

```diff
@@ -388,21 +396,30 @@ export interface ColumnPickerActionsContext {
 
 /** An interface representing a grouped item */
 export interface ColumnPickerGroupItem {
+    // The name of the group that this column belongs to.
     group?: string;
+    // The name of the column.
     name: string;
 }
 
 /** An interface representing settings of groups defined in ColumnPickerGroupItem objects */
 export interface ColumnPickerGroupSetting {
+    // The name of the group this setting object is related to.
     group: string;
+    // Defines if this group will be expanded on load. This is an optional property.
     initiallyExpanded?: boolean;
 }
 
 /** Interface representing a tree node item. This normalises data into one format */
 export class ColumnPickerTreeNode {
```

Comment:
Sorry this should be an interface, I'm not sure why I changed it.

Reply:
@donna-marijne  As for the comment. It was following the pattern of matching what was in the docs but this has been removed from the docs since.

Reply:
I think it's correct as a class since it's only used internally. Clean Code is on OReilly.

Reply:
@donna-marijne Ok sure changed back and the comment updated.

Reply:
The only downside with a class is that it is a runtime construct so will be included in the JavaScript output, whereas an interface is a compile time construct only and not included in any build output, so its slightly better for us to use interfaces unless we need any methods etc.. as we don't increase our bundle size unnecessarily.

Reply:
Ah, I did not think of that. I think there are a few functions which ought to be refactored from `ColumnPickerComponent` to `ColumnPickerTreeNode`. However, realistically speaking we will never come back to refactor this, so it might as well be an interface.